src/catalog/function_catalog.cpp

```diff
@@ -0,0 +1,355 @@
+          "arg_names", false); //val20
+
+
+      auto arg_defaults_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "arg_defaults", false); //21
+
+      auto src_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "src", false); //22
+      name_column.AddConstraint(not_null_constraint);
+
+      auto bin_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "bin", false); //23
+      
+      auto config_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "config", false); //24
+
+      auto aclitem_column = catalog::Column(type::Type::VARCHAR, varchar_type_size,
+          "aclitem", false); //25
+
+      std::unique_ptr<catalog::Schema> function_catalog_schema(
+          new catalog::Schema({function_oid_column, name_column, owner_oid_column,
+            lang_oid_column, cost_column, rows_column, variadic_oid_column,isagg_column, iswindow_column, secdef_column, leakproof_column, isstrict_column,retset_column, volatile_column, num_args_column, num_default_args_column,rettype_oid_column, arg_types_column, all_arg_types_column, arg_modes_column, arg_names_column, arg_defaults_column, src_column, bin_column, config_column, aclitem_column}));
+
+      return function_catalog_schema;
+    }
+
+ UDFFunctionData FunctionCatalog::GetFunction(const std::string &name,
+                                   concurrency::Transaction *txn) {
+  // Write logic to populate the fields of UDFFunctionData
+
+  LOG_DEBUG("GetFunction call to search for udf in pg_proc");
+
+  std::vector<oid_t> column_ids({1,3,16,17,20,22});
+  oid_t index_offset = 1;
+  std::vector<type::Value> values;
+  values.push_back(type::ValueFactory::GetVarcharValue(name,nullptr).Copy());
+ 
+  UDFFunctionData function_info;
+  function_info.func_is_present_ = false;
+
+  auto result_tiles = GetResultWithIndexScan(column_ids, index_offset, values, txn);
+  
+   if (result_tiles->size() != 0) {
+
+    if ((*result_tiles)[0]->GetTupleCount() != 0) {
+      function_info.func_name_ = (*result_tiles)[0]->GetValue(0, 0).ToString();  // After projection left 1 column
+
+      function_info.language_id_ = (*result_tiles)[0]->GetValue(0, 1).GetAs<peloton::PLType>(); 
+
+      function_info.return_type_ = (*result_tiles)[0]
+                      ->GetValue(0, 2)
+                      .GetAs<type::Type::TypeId>();  // After projection left 1 colum
+
+      auto arg_types = (*result_tiles)[0]->GetValue(0, 3).ToString();  // After projection left 1 column
+      std::vector<std::string> arg_types_split;
+      boost::split(arg_types_split,arg_types, boost::is_any_of(" "));
+      
+      for(auto e : arg_types_split){
+        if(e.size() != 0)
+          function_info.argument_types_.push_back(static_cast<type::Type::TypeId>(std::stoi(e)));
+      }
+      
+      auto arg_names = (*result_tiles)[0]->GetValue(0, 4).ToString();  // After projection left 1 column
```

Comment:
Also 4 is magic number here. Maybe it is better to define 4 in header file with some explaination.